### ISSUE TITLE
List::Util 1.45 Required But Not Enforced, Installation Fails

### ISSUE THREAD

**Issue by wbraswell** [Perl5-Alien/Alien-Base-ModuleBuild#16]
Installation of Alien::Base::ModuleBuild fails on a fresh Ubuntu v16.04.4 with the default Perl v5.22.1, this is causing everything else to fail such as Alien::GSL and RPerl itself, etc etc.

The main failure appears to be due to the currently-installed List::Util v1.41 not providing the `uniq()` subroutine:

```
t/00_diag.t .................................. ok
"uniq" is not exported by the List::Util module
Can't continue after import errors at /home/wbraswell/.cpanm/work/1580604471.6262/Alien-Base-ModuleBuild-1.11/blib/lib/Alien/Base/ModuleBuild.pm line 16
```

I can see that we are already requiring List::Util v1.45 in order get access to the 'uniq()' subroutine:

https://github.com/Perl5-Alien/Alien-Base-ModuleBuild/blob/master/dist.ini#L29

However, `cpanm` does not actually enforce the requirement of List::Util v1.45, and lets Alien::Base::ModuleBuild start running tests without recursively installing a List::Util upgrade.

The full output from A::B::MB failure is below:

```
$ cpanm -v --installdeps RPerl
...
Building and testing Alien-Base-ModuleBuild-1.11 ... Building Alien-Base-ModuleBuild
# 
# 
# 
# HARNESS_ACTIVE=1
# HARNESS_VERSION=3.35
# LANG=en_US.UTF-8
# PERL5LIB=/home/wbraswell/.cpanm/work/1580604471.6262/Alien-Base-ModuleBuild-1.11/blib/lib:/home/wbraswell/.cpanm/work/1580604471.6262/Alien-Base-ModuleBuild
-1.11/blib/arch:lib:/home/wbraswell/perl5/lib/perl5/5.22.1/x86_64-linux-gnu-thread-multi:/home/wbraswell/perl5/lib/perl5/5.22.1:/home/wbraswell/perl5/lib/perl
5/x86_64-linux-gnu-thread-multi:/home/wbraswell/perl5/lib/perl5:lib:/home/wbraswell/perl5/lib/perl5
# PERL_DL_NONLAZY=1
# PERL_LOCAL_LIB_ROOT=/home/wbraswell/perl5
# PERL_MB_OPT=--install_base "/home/wbraswell/perl5"
# PERL_MM_OPT=INSTALL_BASE=/home/wbraswell/perl5
# PERL_MM_USE_DEFAULT=1
# PERL_USE_UNSAFE_INC=1
# SHELL=/bin/bash
# 
# 
# 
# PERL5LIB path
# /home/wbraswell/.cpanm/work/1580604471.6262/Alien-Base-ModuleBuild-1.11/blib/lib
# /home/wbraswell/.cpanm/work/1580604471.6262/Alien-Base-ModuleBuild-1.11/blib/arch
# lib
# /home/wbraswell/perl5/lib/perl5/5.22.1/x86_64-linux-gnu-thread-multi
# /home/wbraswell/perl5/lib/perl5/5.22.1
# /home/wbraswell/perl5/lib/perl5/x86_64-linux-gnu-thread-multi
# /home/wbraswell/perl5/lib/perl5
# lib
# /home/wbraswell/perl5/lib/perl5
# 
# 
# 
# perl                    5.022001
# Alien::Base             1.98
# Alien::Base::PkgConfig  1.98
# Archive::Extract        0.86
# Archive::Tar            2.04
# Capture::Tiny           0.48
# File::chdir             0.1010
# HTTP::Tiny              0.054
# Module::Build           0.4216
# Path::Tiny              0.112
# Shell::Config::Generate 0.33
# Shell::Guess            0.09
# Sort::Versions          1.62
# Test2::Mock             0.000129
# Test2::Require::Module  0.000129
# Test2::V0               0.000129
# URI                     1.71
# URI::file               4.21
# parent                  0.232
# 
# 
# 
t/00_diag.t .................................. ok
"uniq" is not exported by the List::Util module
Can't continue after import errors at /home/wbraswell/.cpanm/work/1580604471.6262/Alien-Base-ModuleBuild-1.11/blib/lib/Alien/Base/ModuleBuild.pm line 16.
BEGIN failed--compilation aborted at /home/wbraswell/.cpanm/work/1580604471.6262/Alien-Base-ModuleBuild-1.11/blib/lib/Alien/Base/ModuleBuild.pm line 16.
Compilation failed in require at t/alien_base_modulebuild.t line 4.
BEGIN failed--compilation aborted at t/alien_base_modulebuild.t line 4.
t/alien_base_modulebuild.t ................... 
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 
t/alien_base_modulebuild_cabinet.t ........... ok
t/alien_base_modulebuild_file.t .............. ok
t/alien_base_modulebuild_repository.t ........ ok
t/alien_base_modulebuild_repository_ftp.t .... ok
t/alien_base_modulebuild_repository_http.t ... ok
t/alien_base_modulebuild_repository_local.t .. ok
t/alien_base_modulebuild_utils.t ............. ok
t/inline.t ................................... skipped: Module 'Inline' is not installed
t/inline_cpp.t ............................... skipped: Module 'Inline' is not installed

Test Summary Report
-------------------
t/alien_base_modulebuild.t                 (Wstat: 65280 Tests: 0 Failed: 0)
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
Files=11, Tests=39,  3 wallclock secs ( 0.05 usr  0.03 sys +  2.79 cusr  0.28 csys =  3.15 CPU)
Result: FAIL
Failed 1/11 test programs. 0/39 subtests failed.
! Installing Alien::Base::ModuleBuild failed. See /home/wbraswell/.cpanm/work/1580604471.6262/build.log for details. Retry with --force to force install it.
FAIL
```

**Comment by wbraswell** [comment]
If I simply call `cpanm List::Util` myself, then try again, Alien::Base::ModuleBuild passes all tests and installs just fine.

**Comment by plicease** [comment]
Fixed in 1.12

**Comment by wbraswell** [comment]
Works now, thanks @plicease !  :-)
